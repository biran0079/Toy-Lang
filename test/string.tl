print("12345");
print("hello \"world\"");
print("hello\n\\world");
print("Bi" + " " + "Ran");
a="1";
b=a;
a+="2";
print(a);
print(b);
a=a+"3";
print(a);
print(len(a));
print(a[0]);
print(a[1]);
print(a[2]);
print("345"[2]);
fun f(){
  return "hello";
}
print(f()[3]);
