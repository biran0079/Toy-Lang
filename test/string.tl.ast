[stmts, [call, [id, print], [exps, [string, 12345]]], [call, [id, print], [exps, [string, hello "world"]]], [call, [id, print], [exps, [string, hello
\world]]], [call, [id, print], [exps, [+, [+, [string, Bi], [string,  ]], [string, Ran]]]], [=, [id, a], [string, 1]], [=, [id, b], [id, a]], [+=, [id, a], [string, 2]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [id, b]]], [=, [id, a], [+, [id, a], [string, 3]]], [call, [id, print], [exps, [id, a]]], [call, [id, print], [exps, [call, [id, len], [exps, [id, a]]]]], [call, [id, print], [exps, [list_access, [id, a], [int, 0]]]], [call, [id, print], [exps, [list_access, [id, a], [int, 1]]]], [call, [id, print], [exps, [list_access, [id, a], [int, 2]]]], [call, [id, print], [exps, [list_access, [string, 345], [int, 2]]]], [fun, [id, f], [ids], [stmts, [return, [string, hello]]]], [call, [id, print], [exps, [list_access, [call, [id, f], [exps]], [int, 3]]]]]
