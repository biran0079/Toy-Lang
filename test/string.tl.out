12345
hello "world"
hello
\world
Bi Ran
12
1
123
3
1
2
3
5
l
